From 31b9ee2ea157a1fd506f0c6cc72c7c134e39be79 Mon Sep 17 00:00:00 2001
From: Richard Zhu <Richard.Zhu@freescale.com>
Date: Tue, 31 Mar 2015 10:56:04 +0800
Subject: [PATCH] MLK-10512-2 arm: mcc: add mcc related dts nodes on imx7d

- add mu sema4 device nodes on imx7d dts
- add mcc pingpong/tty tests nodes on imx7d dts
- enable mcc test on imx7d
- specify the top 1MB of the 1GB DDR as mcc share memory
- add 7d sema/mu root clocks

Signed-off-by: Richard Zhu <Richard.Zhu@freescale.com>
---
 arch/arm/boot/dts/imx7d-12x12-arm2.dts |  1 +
 arch/arm/boot/dts/imx7d-sdb.dts        |  1 +
 arch/arm/boot/dts/imx7d.dtsi           | 25 +++++++++++++++++++++++--
 3 files changed, 25 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/imx7d-12x12-arm2.dts b/arch/arm/boot/dts/imx7d-12x12-arm2.dts
index 93e0732..bbdbb1d 100644
--- a/arch/arm/boot/dts/imx7d-12x12-arm2.dts
+++ b/arch/arm/boot/dts/imx7d-12x12-arm2.dts
@@ -132,6 +132,7 @@
 	};
 
 	memory {
+		linux,usable-memory = <0x80000000 0x3ff00000>;
 		reg = <0x80000000 0x80000000>;
 	};
 };
diff --git a/arch/arm/boot/dts/imx7d-sdb.dts b/arch/arm/boot/dts/imx7d-sdb.dts
index bb838d6..97470db 100644
--- a/arch/arm/boot/dts/imx7d-sdb.dts
+++ b/arch/arm/boot/dts/imx7d-sdb.dts
@@ -16,6 +16,7 @@
 	compatible = "fsl,imx7d-sdb", "fsl,imx7d";
 
 	memory {
+		linux,usable-memory = <0x80000000 0x3ff00000>;
 		reg = <0x80000000 0x80000000>;
 	};
 
diff --git a/arch/arm/boot/dts/imx7d.dtsi b/arch/arm/boot/dts/imx7d.dtsi
index 530711d..6fa0714 100644
--- a/arch/arm/boot/dts/imx7d.dtsi
+++ b/arch/arm/boot/dts/imx7d.dtsi
@@ -1056,10 +1056,31 @@
 			};
 
 			mu: mu@30aa0000 {
-				compatible = "fsl,imx7d-mu", "fsl,imx6sx-mu", "syscon";
+				compatible = "fsl,imx7d-mu", "fsl,imx6sx-mu";
 				reg = <0x30aa0000 0x10000>;
 				interrupts = <GIC_SPI 88 IRQ_TYPE_LEVEL_HIGH>;
-				status = "disabled";
+				clocks = <&clks IMX7D_MU_ROOT_CLK>;
+				clock-names = "mu";
+				status = "okay";
+			};
+
+			sema4: sema4@30ac0000 {
+				compatible = "fsl,imx7d-sema4";
+				reg = <0x30ac0000 0x10000>;
+				interrupts = <GIC_SPI 116 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&clks IMX7D_SEMA4_HS_ROOT_CLK>;
+				clock-names = "sema4";
+				status = "okay";
+			};
+
+			mcctest: mcctest{
+				compatible = "fsl,imx6sx-mcc-test";
+				status = "okay";
+			};
+
+			mcctty: mcctty{
+				compatible = "fsl,imx6sx-mcc-tty";
+				status = "okay";
 			};
 
 			usbotg1: usb@30b10000 {
-- 
1.8.0

